eng/common/scripts/Update-ChangeLog.ps1

```diff
@@ -0,0 +1,109 @@
+# Note: This script will add or replace version title in change log
+
+# Parameter description
+# Version : Version to add or replace in change log
+# ChangeLogPath: Path to change log file. If change log path is set to directory then script will probe for change log file in that path 
+# Unreleased: Default is true. If it is set to false, then today's date will be set in verion title. If it is True then title will show "Unreleased"
+# ReplaceVersion: This is useful when replacing current version title with new title.( Helpful to update the title before package release)
+
+param (
+  [Parameter(Mandatory = $true)]
+  [String]$Version,
+  [Parameter(Mandatory = $true)]
+  [String]$ServiceDirectory,
+  [Parameter(Mandatory = $true)]
+  [String]$PackageName,
+  [boolean]$Unreleased=$True,
+  [boolean]$ReplaceVersion = $False
+)
+
+DynamicParam {
+    if ($Unreleased -eq $False) {
+        $releaseStatusAttribute = New-Object System.Management.Automation.ParameterAttribute
+        $releaseStatusAttribute.Mandatory = $False
+        $attributeCollection = New-object System.Collections.ObjectModel.Collection[System.Attribute]
+        $attributeCollection.Add($releaseStatusAttribute)
+        $releaseStatusParam = New-Object System.Management.Automation.RuntimeDefinedParameter('ReleaseDate', [string], $attributeCollection)
+        $paramDictionary = New-Object System.Management.Automation.RuntimeDefinedParameterDictionary
+        $paramDictionary.Add('ReleaseDate', $releaseStatusParam)
+        return $paramDictionary
+    }
+}
+
+Begin {
+    . "${PSScriptRoot}\common.ps1"
+    $UNRELEASED_TAG = "(Unreleased)"
+
+    $dateFormat = "yyyy-MM-dd"
+    $provider = [System.Globalization.CultureInfo]::InvariantCulture
+    if ($ReleaseDate)
```

Comment:
We could always make ReleaseDate be a DateTime parameter and if it set then we can just call ToString on it we shouldn't need to parse it and then call ToString.